qiskit/transpiler/passes/layout/csp_layout.py

```diff
@@ -97,48 +97,64 @@ def __init__(self, coupling_map, strict_direction=False, seed=None, call_limit=1
         self.strict_direction = strict_direction
         self.call_limit = call_limit
         self.time_limit = time_limit
-        self.seed = seed
+        self.seed = seed or 42
```

Comment:
Probably a mistake, if seed is `None` you either should not seed the RNG or seed it with `None`
```suggestion
        self.seed = seed
```